modules/rostests/apitests/ws2_32/gethostname.c

```diff
@@ -0,0 +1,319 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     Tests for Hostname command and Winsock gethostname function
+ * COPYRIGHT:   Copyright 2019 Doug Lyons <douglyons@douglyons.com>
+ *
+ */
+
+#include<stdio.h>
+#include<stdlib.h>
+#include<string.h>
+#include<windows.h>
+#include<winreg.h>
+
+START_TEST(gethostname)
+{
+    CHAR file_name[25] = {"HostName123.txt"};
+    CHAR cmdline[100] = {"hostname > "};
+    FILE *fp;
+    CHAR mystring [100] = {0};
+    CHAR *myreturn;
+    INT len, pos;
+    HKEY hKeyHN;
+    DWORD cbData;
+    LONG Error;
+    CHAR szHostNameOld[MAX_PATH];
+    CHAR szHostNameNew[MAX_PATH] = {0};
+    CHAR hostbuffer[1024] = {0};
+    INT hnerror;
+    INT iResult;
+    WSADATA wsaData;
+
+HKEY OpenHostNameKey(void)
+{
+    static const WCHAR RegHostNameKey[] = L"System\\CurrentControlSet\\Services\\Tcpip\\Parameters";
+    HKEY hKey = NULL;
+    LONG Error = RegOpenKeyExW(HKEY_LOCAL_MACHINE, RegHostNameKey, 0, KEY_ALL_ACCESS, &hKey);
+    if (!Error)
+        return hKey;
+    return NULL;
+}
+
+HKEY OpenComputerNameKey(void)
```

Comment:
Functions cannot be embedded in C.
They should be moved (at the top) as separate functions.